src/utils/network-monitor.ts

```diff
@@ -1447,19 +1474,35 @@ export class NetworkMonitor {
         }
 
         if (this.processTransactions2 !== undefined) {
-          await this.processTransactions2?.bind(this.parent)(job, interestingTransactions)
+          try {
+            await this.processTransactions2?.bind(this.parent)(job, interestingTransactions)
+          } catch (error: any) {
+            this.structuredLogError(job.network, `Error processing transactions ${error}`, job.block)
+          }
         }
 
-        this.blockJobHandler(job.network, job)
+        try {
+          return await this.blockJobHandler(job.network, job)
```

Comment:
we have this method `blockJobHandler` called a few different times based on this if else tree. I wonder if we can organize it so it is only called once.